### PR TITLE
Allow FMU start time to be specified

### PR DESCRIPTION
**Pull request by nllong**
This change ensures that the starttime (in seconds) is respected in the FMU. See comments below.

### CODE REVIEW

File: worker/common/data/categories.json [hunk 1]
```diff
@@ -1,21 +1,53 @@
...
-"emissions": ["EmissionsElectricPower",
-              "EmissionsGasPower",
-              "EmissionsDistrictHeatingPower",
-              "EmissionsBiomassPower",
-              "EmissionsSolarThermalPower"],
-"occupancy": ["Occupancy"],
-"internalGains": ["InternalGains"],
-"setpoints": ["LowerSetp", "UpperSetp"]}
\ No newline at end of file
+{
```
Comment (nllong) [inline worker/common/data/categories.json]: Pretty formatting.

File: worker/common/testcase.py [hunk 2]
```diff
@@ -32,16 +32,30 @@
...
-        
+
+    def __init__(self, **kwargs):
+        """
+        Constructor
+
+        :param kwargs:
+        """
+        # ensure the fmupath and step are defined in kwargs
+        if kwargs.get('fmupath', None) is None:
```
Comment (nllong) [inline worker/common/testcase.py]: Use kwargs to pass the options. Default to start_time of zero.

File: worker/common/testcase.py [hunk 3]
```diff
@@ -69,31 +83,31 @@ def __init__(self, con):
...
         # Set default options
         self.options = self.fmu.simulate_options()
-        self.options['CVode_options']['rtol'] = 1e-6 
+        self.options['CVode_options']['rtol'] = 1e-6
         # Set default communication step
-        self.set_step(con['step'])
+        self.set_step(init_options['step'])
         # Set initial simulation start
-        self.start_time = 0
+        self.start_time = init_options['start_time']
```
Comment (nllong) [inline worker/common/testcase.py]: note that this was previously 0

File: worker/stepfmu/step_fmu.py [hunk 4]
```diff
@@ -88,18 +88,19 @@ def __init__(self, **kwargs):
...
-            
+
         (self.tagid_and_outputs, self.id_and_dis, self.default_input) = self.create_tag_dictionaries(tagpath)
- 
-        #initiate the testcase
-        self.tc = common.testcase.TestCase(config) 
-        
+
+        #initiate the testcase. Pass the config dict as arguments to use the kwargs in the initializer.
+        self.tc = common.testcase.TestCase(**config)
```
Comment (nllong) [inline worker/stepfmu/step_fmu.py]: flatten the dict to arg value pairs

File: worker/stepfmu/step_fmu.py [hunk 5]
```diff
@@ -259,19 +260,19 @@ def step(self):
...
 if real_time_flag:
-    time_scale = 1 
+    time_scale = 1
 startTime = sys.argv[4]
 if startTime == 'undefined':
-    startTime = 0;
+    startTime = 0
 else:
-    starTime = int(sys.argv[4])
+    startTime = int(sys.argv[4])
```
Comment (nllong) [inline worker/stepfmu/step_fmu.py]: Yeah, this one took awhile to find! starTime -> startTime

File: alfalfa_worker/step_sim/step_fmu.py [hunk 6]
```diff
@@ -265,7 +266,7 @@ def step(self):
 if startTime == 'undefined':
     startTime = 0
 else:
-    starTime = int(sys.argv[4])
+    startTime = int(sys.argv[4])
```
Comment (nllong) [inline alfalfa_worker/step_sim/step_fmu.py:269]: this is part 2 of the main update.

File: alfalfa_worker/lib/testcase.py [hunk 7]
```diff
@@ -72,9 +87,9 @@ def __init__(self, con):
         self.options = self.fmu.simulate_options()
         self.options['CVode_options']['rtol'] = 1e-6
         # Set default communication step
-        self.set_step(con['step'])
+        self.set_step(init_options['step'])
         # Set initial simulation start
-        self.start_time = 0
+        self.start_time = init_options['start_time']
```
Comment (nllong) [inline alfalfa_worker/lib/testcase.py:92]: this is part 1 of the main update

**Comment by corymosiman12** [comment]
Not much work has been done in the fmu step_sim section, so this seems all good

**Pull request merged**